src/utils.jl

```diff
@@ -56,11 +58,51 @@ end
 
 # family of projection functions
 
-immutable Proj{field}; end
+immutable Proj{F}
+    f::F
+end
+
+(p::Proj)(x::Tup) = p.f(x)
+function (p::Proj{Val{f}}){f}(x::Tup)
+    getfield(x, f)
+end
 
-(::Proj{field}){field}(x) = getfield(x, field)
+"""
+    @pick(fields...)
+
+Returns a callable object `f` such that `f(x::Tuple)` returns a `Tuple` with only
+elements of index specified by `fields`, `f(x::NamedTuple)` return a `Tuple` if
+`fields` are integers, or a `NamedTuple` if `fields` are symbols with only the specified
+fields in the output.
+
+The callable is specialized to work efficiently on `Columns` by calling it once
+on `.columns` field to get the equivalent result.
+
+Calling `map` on an `IndexedTable` with a `@pick` callable will run the callable on
+the data columns.
+
+# Examples
+    c = Columns(x=[1], y=[2.0])
+    @pick(2)(c) == Columns([2.0])
+    @pick(y)(c) == Columns(y=[2.0])
+    @pick(2,1)(c) == Columns([2.0], [1])
+    @pick(y,x)(c) == Columns(y=[2.0], x=[1])
+
+    t = IndexedTable([1], c)
+    map(@pick(y, x), t) == IndexedTables([1], Columns(y=[2.0], x=[1]))
+"""
+macro pick(ex...)
+    tup = if all([isa(x, Symbol) for x in ex])
+        # Named tuple
+        T = NamedTuples.create_tuple([x for x in ex])
+        :(NamedTuples.$T($([:(getfield(x, $(Expr(:quote, f)))) for f in ex]...)))
```

Comment:
Oops, you're right. We should actually try to rely on the `serialize` method we've added for NamedTuple types (which is in JuliaDB, but should be moved to NamedTuples).

It would also be nice if this macro could do less work. Ideally it should expand to a `@NT(...)` expression, i.e. expand to exactly what you'd write if the macro didn't exist, instead of using NamedTuple internals.

To make all that work, we could have NamedTuples put the type it creates directly into its macro output, instead of generating a GlobalRef.